src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -2737,26 +2737,43 @@ private static int IndexOfMinMaxCore<TIndexOfMinMax>(ReadOnlySpan<float> x)
             // otherwise returns the index of the greater of the inputs.
             // It treats +0 as greater than -0 as per the specification.
 
-            if (Vector512.IsHardwareAccelerated && x.Length >= Vector512<float>.Count)
+            if (Vector512.IsHardwareAccelerated && Vector512<T>.IsSupported && x.Length >= Vector512<T>.Count)
             {
-                ref float xRef = ref MemoryMarshal.GetReference(x);
-                Vector512<int> resultIndex = Vector512.Create(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
-                Vector512<int> curIndex = resultIndex;
-                Vector512<int> increment = Vector512.Create(Vector512<float>.Count);
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                static Vector512<T> CreateVector512T(int i) =>
+                    sizeof(T) == sizeof(long) ? Vector512.Create((long)i).As<long, T>() :
+                    sizeof(T) == sizeof(int) ? Vector512.Create(i).As<int, T>() :
+                    sizeof(T) == sizeof(short) ? Vector512.Create((short)i).As<short, T>() :
+                    sizeof(T) == sizeof(byte) ? Vector512.Create((byte)i).As<byte, T>() :
+                    throw new NotSupportedException();
```

Comment:
Why not just `Vector512.Create(T.CreateTruncating(i))`?

Reply:
Heh, because I forgot about it :)

Reply:
Oh, actually, no, this is because I don't want T, I want the integer types, then reinterpreted as T.  And `Vector128.Create(1).AsSingle()` is different from `Vector128.Create(1f)`.

Reply:
Ah right, unfortunate there isn't an easier way to do this.

For the new vectorization logic I've been using ISimdVector and taking multiple constrained generic parameters

Not possible while this is an OOB unfortunately